app/code/community/Razorpay/Payments/Model/Paymentmethod.php

```diff
@@ -107,14 +112,15 @@ public function captureOrder(Varien_Object $order, $amount)
             $error = $e->getMessage();
         }
 
-        return [$result, $error];
+        return array($result, $error);
     }
 
     public function validateSignature($response)
     {
-        $requestFields = Mage::app()->getRequest()->getPost();
-
-        $paymentId = $requestFields['razorpay_payment_id'];
+        if (isset($response['razorpay_payment_id']) === false)
+        {
+            throw new Exception('Razorpay Payment Id is not defined');
```

Comment:
Same thing here as well.

Maybe have a single validation function before that validates the entire $response array instead of this?